test/distributed/optim/test_zero_redundancy_optimizer.py

```diff
@@ -204,6 +204,47 @@ def test_zero_grad(self):
         self.assertFalse(m.weight.grad)
         self.assertFalse(m.bias.grad)
 
+    def test_constructor(self):
+        """Check the robustness of the ZeroRedundancyOptimizer constructor by
+        passing different values for `params`"""
+        self.dist_init(self.rank)
+
+        m = torch.nn.Linear(1, 1)
+        # (input, expected error)
+        inputs = [
+            ([], ValueError),                           # empty parameter list
+            (torch.randn(1), TypeError),                # non-iterable: `torch.Tensor`
+            (1.2, TypeError),                           # non-iterable: `float`
+            ([{"params": m.parameters()}], TypeError),  # iterable of dict
+            (list(m.parameters()) + [42], TypeError),   # iterable containing non-`torch.Tensor`
+            (m.parameters(), None),                     # `params` as a generator
+            (list(m.parameters()), None)                # `params` as a list
+        ]
+
+        for (input, error) in inputs:
```

Comment:
nit: parentheses around `input, error` and `error` not necessary.

Reply:
Fixed.